src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormattingContext.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
+{
+    /// <summary>
+    /// Represents an object containing the information to serialize with <see cref="IDistributedCacheTagHelperFormatter" />.
+    /// </summary>
+    public class DistributedCacheTagHelperFormattingContext
```

Comment:
@lodejard how do you feel about either one of these alternatives:

1. Remove this entirely and bake in the `Key` and `Html` into the `IDistributedCacheTagHelperFormatter.SerializeAsync` signature.
1. Add a constructor to this class and force the user to provide the `Html`/`Key`. Would mean we remove the setter on the properties here.

Both of these scenarios would make it overwhelmingly clear which parameters are required to properly cache something in a distributed cache without the need to do ugly bits like `if (context.Html == null) { throw ... }` in the `IDistributedCacheTagHelperFormatter.SerializeAsync` impl